enhancements/microshift/microshift-supportability-tools.md

```diff
@@ -0,0 +1,397 @@
+  preloaded when installing MicroShift. Either option is not the best when it
+  comes to MicroShift.
+* OCP tie. Since `must-gather` was specifically made for OCP, it is tailored to
+  run in such solution. There are many specific resources and system components
+  that MicroShift does not have, triggering errors. Some of the relevant system
+  components for MicroShift are not checked in `must-gather` because they have
+  a different name, dont exist, or don't exist.
+* Adapting `must-gather` to run in MicroShift is not a trivial task, as it not
+  only requires MicroShift awareness, but also violates some of the
+  MicroShift's principles, like lightness. It would be debatable if the tool
+  should have such intelligence/complexity.
+
+It is desirable to have the MicroShift tool match the output format of
+`must-gather`, as there are several tools easing debuggability:
+* [omg](https://github.com/kxr/o-must-gather)
+* [omc](https://github.com/gmeghnag/omc)
+
+Plus, CEE is used to this format when it comes to reports.
+
+### Sos
+[sos](https://github.com/sosreport/sos) is a data collection tool targetting
+any UNIX based system. It is composed of specialized plugins which gather all
+the relevant data from the different components and applications in a system.
+
+By executing `sos report` it will generate a detailed report with all the
+enabled plugins (listed by running `sos report -l`).
+
+A [plugin](https://github.com/sosreport/sos/wiki/How-to-Write-a-Plugin) is a
+python script included as part of `sos` and makes the tool extensible. Plugins
+also have an enablement check to run by themselves upon a report command, or
+may be invoked manually.
+
+Among the plugins already available we can find one for [OpenShift](https://github.com/sosreport/sos/blob/main/sos/report/plugins/openshift.py).
+This plugin relies on the usage of `oc` command line tool. It gathers data from
+OpenShift API resources and collects logs for pods and kubelet. While
+MicroShift shares similarities with this plugin, it is not directly usable for
+several reasons:
+* Using `oc` tool means it needs to be installed in the host. As seen before,
+  MicroShift strives for lightness and this package is not installed. Doing so
+  needs ~125MB.
+* Retrieves specific OpenShift API resources that are not present in MicroShift.
+
+Since none of the available plugins fulfills MicroShift's needs, we will need a
+new one.
+
+Requirements for the `sos` plugin follow:
+* Not require usage of other tools than what is already present in the host.
+  Introduce any of the `oc` commands in `microshift` binary. `microshift`
+  already supports a `get` command.
+* Have `must-gather` format to use already existing debug tools.
+* Retrieve MicroShift specific resources only, not covered by other plugins.
+  These include cluster resources, MicroShift configuration, MicroShift
+  version and MicroShift logs. Some specifics, like ovnk might require an
```

Comment:
Do we know if OVN-K will require a separate plugin?

Reply:
Yes, like OpenShift plugin does.